api/sys_plugins.go

```diff
@@ -225,6 +225,34 @@ func (c *Sys) DeregisterPlugin(i *DeregisterPluginInput) error {
 	return err
 }
 
+// ReloadPluginInput is used as input fo the ReloadPlugin function.
+type ReloadPluginInput struct {
+	// Plugin is the name of the plugin to reload, as registered in the plugin catalog
+	Plugin string `json:"plugin"`
+
+	// Mounts is the array of string mount paths of the plugin backends to reload
+	Mounts []string `json:"mounts"`
+}
+
+// ReloadPlugin reloads mounted plugin backends
+func (c *Sys) ReloadPlugin(i *ReloadPluginInput) error {
+	path := "/v1/sys/plugins/reload/backend"
+	req := c.c.NewRequest(http.MethodPut, path)
+
+	if err := req.SetJSONBody(i); err != nil {
+		return err
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	resp, err := c.c.RawRequestWithContext(ctx, req)
```

Comment:
I would do it this way.

```
	resp, err := c.c.RawRequestWithContext(ctx, req)
	if err != nil {
		return err
	}
	defer resp.Body.Close()
	return nil
```

Reply:
There are a couple of places that does it the way you have done it. But given a choice, I would modify those too instead of adding more such instances.

Reply:
I agree, the proposed snippet is more obvious